### ISSUE TITLE
Big term in a query breaks highlighting

### ISSUE THREAD

**Issue by radoondas** [elastic/elasticsearch#46914]
**Elasticsearch version** (`bin/elasticsearch --version`): 
```
"version" : {
    "number" : "6.5.1",
    "build_flavor" : "default",
    "build_type" : "rpm",
    "build_hash" : "8c58350",
    "build_date" : "2018-11-16T02:22:42.182257Z",
    "build_snapshot" : false,
    "lucene_version" : "7.5.0",
    "minimum_wire_compatibility_version" : "5.6.0",
    "minimum_index_compatibility_version" : "5.0.0"
  }
```

**Plugins installed**: [discovery-ec2]

**JVM version** (`java -version`): OpenJDK 1.8.0_222

**OS version** (`uname -a` if on a Unix-like system): Linux <hostname> 3.10.0-957.27.2.el7.x86_64 #1 SMP Mon Jul 29 17:46:05 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

**Description of the problem including expected versus actual behavior**:
When a query has big term it breaks highlighting.

**Steps to reproduce**:
Use long term in query - more than 1000 characters.

**Provide logs (if relevant)**:
```
[2019-09-19T16:40:25,437][DEBUG][o.e.a.s.TransportSearchAction] [hostname] [60171807] Failed to execute fetch phase
org.elasticsearch.transport.RemoteTransportException: [hostname][127.0.0.1:9300][indices:data/read/search[phase/fetch/id]]
Caused by: java.lang.IllegalArgumentException: This builder doesn't allow terms that are larger than 1,000 characters, got CPU OS 12_3_1 like Mac OS X iPad AppleWebKit/605.1 Mobile/15E148 Safari/605.1.15 (AirWatch Browser v7.8 Safari/605.1.15 (AirWatch Browser v7.8 Safari/605.1.15 (AirWatch Browser v7.8 Safari/605.1.15 (AirWatch Browser v7.8 Safari/605.1.15 (AirWatch Browser v7.8 Safari/605.1.15 (AirWatch Browser v7.8 Safari/605.1.15 (AirWatch Browser v7.8 Safari/605.1.15 (AirWatch Browser v7.8 Safari/605.1.15 (AirWatch Browser v7.8 Safari/605.1.15 (AirWatch Browser v7.8 Safari/605.1.15 (AirWatch Browser v7.8 Safari/605.1.15 (AirWatch Browser v7.8 Safari/605.1.15 (AirWatch Browser v7.8 Safari/605.1.15 (AirWatch Browser v7.8 Safari/605.1.15 (AirWatch Browser v7.8 Safari/605.1.15 (AirWatch Browser v7.8 Safari/605.1.15 (AirWatch Browser v7.8 Safari/605.1.15 (AirWatch Browser v7.8 Safari/605.1.15 (AirWatch Browser v7.8 Safari/605.1.15 (AirWatch Browser v7.8 Version/12.3 Version/12.3 Version/12.3 Version/12.3 Version/12.3 Version/12.3 Version/12.3 Version/12.3 Version/12.3 Version/12.3 Version/12.3 Version/12.3 Version/12.3 Version/12.3 Version/12.3 Version/12.3 Version/12.3 Version/12.3 Version/12.3 Version/12.3
at org.apache.lucene.util.automaton.DaciukMihovAutomatonBuilder.add(DaciukMihovAutomatonBuilder.java:231) ~[lucene-core-7.5.0.jar:7.5.0 b5bf70b7e32d7ddd9742cc821d471c5fabd4e3df - jimczi - 2018-09-18 13:01:13]
at org.apache.lucene.util.automaton.DaciukMihovAutomatonBuilder.build(DaciukMihovAutomatonBuilder.java:306) ~[lucene-core-7.5.0.jar:7.5.0 b5bf70b7e32d7ddd9742cc821d471c5fabd4e3df - jimczi - 2018-09-18 13:01:13]
at org.apache.lucene.util.automaton.Automata.makeStringUnion(Automata.java:581) ~[lucene-core-7.5.0.jar:7.5.0 b5bf70b7e32d7ddd9742cc821d471c5fabd4e3df - jimczi - 2018-09-18 13:01:13]
at org.apache.lucene.search.uhighlight.MemoryIndexOffsetStrategy.buildCombinedAutomaton(MemoryIndexOffsetStrategy.java:67) ~[lucene-highlighter-7.5.0.jar:7.5.0 b5bf70b7e32d7ddd9742cc821d471c5fabd4e3df - jimczi - 2018-09-18 13:02:19]
at org.apache.lucene.search.uhighlight.MemoryIndexOffsetStrategy.<init>(MemoryIndexOffsetStrategy.java:57) ~[lucene-highlighter-7.5.0.jar:7.5.0 b5bf70b7e32d7ddd9742cc821d471c5fabd4e3df - jimczi - 2018-09-18 13:02:19]
at org.apache.lucene.search.uhighlight.UnifiedHighlighter.getOffsetStrategy(UnifiedHighlighter.java:866) ~[lucene-highlighter-7.5.0.jar:7.5.0 b5bf70b7e32d7ddd9742cc821d471c5fabd4e3df - jimczi - 2018-09-18 13:02:19]
at org.apache.lucene.search.uhighlight.CustomUnifiedHighlighter.getFieldHighlighter(CustomUnifiedHighlighter.java:149) ~[elasticsearch-6.5.1.jar:7.5.0 b5bf70b7e32d7ddd9742cc821d471c5fabd4e3df - jimczi - 2018-09-18 13:02:19]
at org.apache.lucene.search.uhighlight.UnifiedHighlighter.highlightFieldsAsObjects(UnifiedHighlighter.java:574) ~[lucene-highlighter-7.5.0.jar:7.5.0 b5bf70b7e32d7ddd9742cc821d471c5fabd4e3df - jimczi - 2018-09-18 13:02:19]
at org.apache.lucene.search.uhighlight.CustomUnifiedHighlighter.highlightField(CustomUnifiedHighlighter.java:107) ~[elasticsearch-6.5.1.jar:7.5.0 b5bf70b7e32d7ddd9742cc821d471c5fabd4e3df - jimczi - 2018-09-18 13:02:19]
at org.elasticsearch.search.fetch.subphase.highlight.UnifiedHighlighter.highlight(UnifiedHighlighter.java:128) ~[elasticsearch-6.5.1.jar:6.5.1]
at org.elasticsearch.search.fetch.subphase.highlight.HighlightPhase.hitExecute(HighlightPhase.java:110) ~[elasticsearch-6.5.1.jar:6.5.1]
at org.elasticsearch.search.fetch.FetchPhase.execute(FetchPhase.java:159) ~[elasticsearch-6.5.1.jar:6.5.1]
at org.elasticsearch.search.SearchService.lambda$executeFetchPhase$3(SearchService.java:556) ~[elasticsearch-6.5.1.jar:6.5.1]
at org.elasticsearch.search.SearchService$3.doRun(SearchService.java:361) ~[elasticsearch-6.5.1.jar:6.5.1]
at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:723) ~[elasticsearch-6.5.1.jar:6.5.1]
at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) ~[elasticsearch-6.5.1.jar:6.5.1]
at org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:41) ~[elasticsearch-6.5.1.jar:6.5.1]
at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) ~[elasticsearch-6.5.1.jar:6.5.1]
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[?:1.8.0_222]
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[?:1.8.0_222]
at java.lang.Thread.run(Thread.java:748) [?:1.8.0_222]
```

**Comment by Dchamard** [comment]
Running into the same problem.

**Comment by romseygeek** [comment]
This is a lucene issue, I've filed https://github.com/apache/lucene/pull/11808 to fix

**Comment by romseygeek** [comment]
The fix was included in the lucene 9.5 release, incorporated into elasticsearch 8.7
